psl-core/src/main/java/org/linqs/psl/application/inference/InferenceApplication.java

```diff
@@ -109,14 +114,39 @@ protected void initialize() {
      */
     protected abstract void completeInitialize();
 
+    /**
+     * Minimizes the total weighted incompatibility of the GroundAtoms in the Database
+     * according to the Model and commits the updated truth values back to the Database.
+     */
+    public void inference() {
+        inference(false);
+    }
+
     /**
      * Minimizes the total weighted incompatibility of the GroundAtoms in the Database
      * according to the Model and commits the updated truth values back to the Database.
      *
      * The RandomVariableAtoms to be inferred are those persisted in the Database when this method is called.
      * All RandomVariableAtoms which the Model might access must be persisted in the Database.
      */
-    public abstract void inference();
+    public void inference(boolean skipAtomCommit) {
```

Comment:
Swap `skipAtomCommit` for `commitAtoms`.
I think it is more intuitive for people to pass `true` when they want something to be done.